s3_upload/utils/log.py

```
@@ -35,29 +39,81 @@ def set_file_handler(logger, log_dir) -> logging.Logger:
     logger.info(
-        "Initialised log fileHandler, setting log output to %s", log_file
+        "Initialised log FileHandler, setting log output to %s", log_file
     )
 
-    check_write_permission_to_log_dir(log_dir)
+    clear_old_logs(logger=logger, log_dir=log_dir, backup_count=5)
+
+    return logger
+
+
+def clear_old_logs(logger, log_dir, backup_count) -> None:
```

Ruff (0.8.0)

Missing type annotation for function argument `logger`

(ANN001)